include/pmacc/particles/memory/dataTypes/Particle.hpp

```diff
@@ -209,19 +203,17 @@ namespace pmacc
                     using SrcTypeSeq = typename Src::ValueTypeSeq;
 
                     /* create attribute list with a subset of common attributes in two sequences
-                     * bmpl::contains has lower complexity than traits::HasIdentifier
+                     * mp_contains has lower complexity than traits::HasIdentifier
                      * and was used for this reason
                      */
-                    using CommonTypeSeq = typename bmpl::copy_if<
-                        DestTypeSeq,
-                        bmpl::contains<SrcTypeSeq, bmpl::_1>,
-                        bmpl::back_inserter<bmpl::vector0<>>>::type;
+#if BOOST_COMP_NVCC && BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(11, 3, 0)
+                    using CommonTypeSeq = mp_copy_if_q<DestTypeSeq, mp_bind_front<mp_contains, SrcTypeSeq>>;
```

Comment:
Why is this workaround required? I am wondering because `mp_set_difference` is working without any issues.
We could rewrite `mp_set_intersection` as

```C++
using UniqueInDestTypeSeq = mp_set_difference<DestTypeSeq, SrcTypeSeq>;
using CommonTypeSeq = mp_set_difference<DestTypeSeq, UniqueInDestTypeSeq>;
```
This would remove the `#if`

Reply:
I created the workaround because nvcc < 11.3 produced a compile error inside `mp_set_intersection`. I can try your workaround.

Reply:
Works.